core/object/object.cpp

```diff
@@ -892,7 +892,7 @@ void Object::set_meta(const StringName &p_name, const Variant &p_value) {
 	if (E) {
 		E->value = p_value;
 	} else {
-		ERR_FAIL_COND(!p_name.operator String().is_valid_identifier());
+		ERR_FAIL_COND_MSG(!p_name.operator String().is_valid_identifier(), "Invalid metadata identifier: '" + p_name + "'.");
```

Comment:
```suggestion
		ERR_FAIL_COND_MSG(!p_name.operator String().is_valid_identifier(), vformat("Invalid metadata identifier: '%s'.", p_name));
```

We use `vformat` to format strings.

Reply:
I have been overruled on this.